### PR TITLE
 sessionctx: refine sql_mode no_backslash_escapes, to pass ci

### PR DESCRIPTION
**Pull request by winkyao**
refer #5104 

@shenli @coocood @tiancaiamao PTAL

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -647,9 +647,11 @@ func (s *session) GetGlobalSysVar(name string) (string, error) {
 func (s *session) SetGlobalSysVar(name string, value string) error {
 	if name == variable.SQLModeVar {
 		value = mysql.FormatSQLModeStr(value)
-		if _, err := mysql.GetSQLMode(value); err != nil {
+		sqlMode, err := mysql.GetSQLMode(value)
```
Comment (coocood) [inline session.go]: We should not update the session when updating global.

Comment (breeswish) [reply session.go]: Evaluate the "final" server status each time when sending the packet?

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
Do we need to add the test of `insert into t value("foo \x00\n\r\x1a\"'\\");` ?

**Comment by winkyao** [comment]
@zimulala the test not easy to add. because it is depend on client behavior.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**